components/script/dom/htmlinputelement.rs

```diff
@@ -111,10 +133,48 @@ struct InputTypeColorShadowTree {
     color_value: Dom<HTMLDivElement>,
 }
 
+// FIXME: These styles should be inside UA stylesheet, but it is not possible without internal pseudo element support.
+const TEXT_TREE_STYLE: &str = "
+#input-editor::selection, #input-placeholder::selection {
+    background: rgba(176, 214, 255, 1.0);
+    color: black;
+}
+
+:host:not(:placeholder-shown) #input-placeholder {
+    visibility: hidden !important
+}
+
+#input-container {
+    position: relative !important;
```

Comment:
We can drop `!important` if no external CSS can match this element.

I think only the placeholder can be matched via `::placeholder` and therefore needs `!important`

Reply:
Sure, thanks!